File: backend/src/main/java/com/example/backend/service/BiologicalProcessService.java
package com.example.backend.service;

import com.example.backend.model.BiologicalProcess;
import com.example.backend.model.exceptions.BiologicalProcessDoesNotExist;

import java.util.Optional;

public interface BiologicalProcessService {
    Optional<BiologicalProcess> getDetailsAboutBiologicalProcess(String biologicalProcessName) throws BiologicalProcessDoesNotExist;
    String getBiologicalProcessAbstract();
}

File: frontend/src/services/BiologicalProcessService.js
import axios from "../custom-axios/axios"

const BiologicalProcessService={
    getBiologicalProcessAbstract:()=>{
        return axios.get('/biologicalProcess')
    },
    getDetailsAboutBiologicalProcess:(name)=>{
        return axios.get(`/biologicalProcess/${name}`)
    }
}
export default BiologicalProcessService;